test/test_logging_interface.cpp

```diff
@@ -0,0 +1,409 @@
+          // is another file with the same prefix but a newer timestamp.
+        }
+      }
+    }
+
+    return found;
+  }
+
+  rcutils_allocator_t allocator;
+
+private:
+  std::string get_expected_log_prefix(const char * name)
+  {
+    char * exe_name;
+    if (name == nullptr || name[0] == '\0') {
+      exe_name = rcutils_get_executable_name(allocator);
+    } else {
+      exe_name = rcutils_strdup(name, allocator);
+    }
+    if (nullptr == exe_name) {
+      throw std::runtime_error("Failed to determine executable name");
+    }
+    std::stringstream prefix;
+    prefix << exe_name << "_" << rcutils_get_pid() << "_";
+    allocator.deallocate(exe_name, allocator.state);
+    return prefix.str();
+  }
+
+  std::filesystem::path test_log_dir_;
+  std::filesystem::path var_log_;
+};
+
+TEST_F(AllocatorTest, init_invalid)
+{
+  EXPECT_EQ(
+    RCL_LOGGING_RET_ERROR,
+    rcl_logging_external_initialize("file_name_prefix", nullptr, bad_allocator));
+  rcutils_reset_error();
+  EXPECT_EQ(
+    RCL_LOGGING_RET_INVALID_ARGUMENT,
+    rcl_logging_external_initialize(nullptr, nullptr, invalid_allocator));
+  rcutils_reset_error();
+}
+
+TEST_F(AllocatorTest, init_failure)
+{
+  RestoreEnvVar home_var("HOME");
+  RestoreEnvVar userprofile_var("USERPROFILE");
+
+  // No home directory to write log to
+  ASSERT_TRUE(rcpputils::set_env_var("HOME", nullptr));
+  ASSERT_TRUE(rcpputils::set_env_var("USERPROFILE", nullptr));
+  EXPECT_EQ(RCL_LOGGING_RET_ERROR, rcl_logging_external_initialize(nullptr, nullptr, allocator));
+  rcutils_reset_error();
+}
+
+TEST_F(AllocatorTest, init_valid)
+{
+  // This success test must be placed in the end after failure cases
+  // with test fixture class AllocatorTest. Because this succeeds,
+  // syslog log facility is allocated, this means init returns always success.
+
+  // Config files are not supported, and pass through with warning.
+  EXPECT_EQ(
+    RCL_LOGGING_RET_OK,
+    rcl_logging_external_initialize(nullptr, "config_file", allocator));
+  rcutils_reset_error();
+}
+
+TEST_F(LoggingTest, log_file_name_prefix)
+{
+  std::string log_file_path;
+  // executable name in default
+  {
+    ASSERT_EQ(RCL_LOGGING_RET_OK, rcl_logging_external_initialize(nullptr, nullptr, allocator));
+    write_something();
+    EXPECT_EQ(RCL_LOGGING_RET_OK, rcl_logging_external_shutdown());
+    EXPECT_NO_THROW(log_file_path = find_single_log(nullptr).string());
+    EXPECT_TRUE(remove_test_folder());
+  }
+  // falls back to executable name if not nullptr, but empty
+  {
+    ASSERT_EQ(RCL_LOGGING_RET_OK, rcl_logging_external_initialize("", nullptr, allocator));
+    write_something();
+    EXPECT_EQ(RCL_LOGGING_RET_OK, rcl_logging_external_shutdown());
+    EXPECT_NO_THROW(log_file_path = find_single_log(nullptr).string());
+    EXPECT_TRUE(remove_test_folder());
+  }
+  // specified by user application
+  {
+    ASSERT_EQ(RCL_LOGGING_RET_OK, rcl_logging_external_initialize("logger", nullptr, allocator));
```

Comment:
test implementation needs to be adjusted because of ABI/API change from jazzy during backport to iron and humble.